python/packages/ai/teams/ai/augmentations/tools_augmentation.py

```diff
@@ -0,0 +1,198 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from __future__ import annotations
+
+import json
+from typing import Dict, List, Optional, Union, cast
+
+from botbuilder.core import TurnContext
+from openai.types import chat
+
+from ...state import MemoryBase
+from ..models.chat_completion_action import ChatCompletionAction
+from ..models.prompt_response import PromptResponse
+from ..planners.plan import Plan, PredictedDoCommand, PredictedSayCommand
+from ..prompts.message import Message
+from ..prompts.sections.prompt_section import PromptSection
+from ..tokenizers.tokenizer import Tokenizer
+from ..validators.validation import Validation
+from .augmentation import Augmentation
+from .tools_constants import (
+    SUBMIT_TOOL_HISTORY,
+    SUBMIT_TOOL_OUTPUTS_MAP,
+    SUBMIT_TOOL_OUTPUTS_MESSAGES,
+    SUBMIT_TOOL_OUTPUTS_VARIABLE,
+)
+
+
+class ToolsAugmentation(Augmentation[Union[str, List[chat.ChatCompletionMessageToolCall]]]):
+    """
+    A server-side 'tools' augmentation.
+    """
+
+    _actions: List[ChatCompletionAction] = []
+
+    def __init__(self, actions: Optional[List[ChatCompletionAction]] = None) -> None:
```

Comment:
NIT: instead of using an optional and an if statement here, it would be cleaner for the argument to default to an empty list so you can just assign the variable without an if statement.